messaging/src/main/java/io/lantern/messaging/Messaging.kt

```diff
@@ -343,14 +347,26 @@ class Messaging(
         }
         replyToSenderId?.let { msgBuilder.setReplyToSenderId(it) }
         replyToId?.let { msgBuilder.setReplyToId(it) }
-        val out =
-            Model.OutboundMessage.newBuilder().setMessageId(base32Id)
+        var out: Model.OutboundMessage.Builder? = null
+        if (sendingToSelf) {
+            msgBuilder.status = Model.StoredMessage.DeliveryStatus.COMPLETELY_SENT
+            msgBuilder.direction = Model.MessageDirection.IN
+        } else {
+            out = Model.OutboundMessage.newBuilder().setMessageId(base32Id)
                 .setSent(sent)
                 .setSenderId(myId.id)
                 .setRecipientId(recipientId)
+        }
         var attachmentId = 0
         attachments?.forEach { attachment ->
-            msgBuilder.putAttachments(attachmentId, attachment)
+            var myAttachment: Model.StoredAttachment.Builder? = null
+            if (sendingToSelf) {
+                myAttachment = attachment.toBuilder()
+                myAttachment.status = Model.StoredAttachment.Status.DONE
+                encryptAttachment(myAttachment)
```

Comment:
We don't want to do this inside here because that will block the calling thread. I'll fix.